discovery-provider/src/premium_content/premium_content_access_checker.py

```diff
@@ -0,0 +1,104 @@
+import json
+import logging
+from typing import Dict, Optional, Tuple, cast
+
+from src.models.tracks.track import Track
+from src.premium_content.helpers import does_user_have_nft_collection
+from src.premium_content.types import PremiumContentType
+from src.utils import db_session
+
+logger = logging.getLogger(__name__)
+
+
+class PremiumContentAccessChecker:
+    # check if content is premium
+    # - if not, then user has access
+    # - if so, check whether user fulfills the conditions
+    #
+    # Returns a tuple of (bool, bool) -> (track is premium, user has access)
+    #
+    # Note: premium content id for type should exist, but just in case it does not,
+    # we return true for user access so that (non-existent) track does not change
+    # existing flow of the function caller.
+    def check_access(
+        self,
+        user_id: int,
+        premium_content_id: int,
+        premium_content_type: PremiumContentType,
+    ) -> Tuple[bool, bool]:
+        # for now, we only allow tracks to be premium
+        # premium playlists will come later
+        if premium_content_type != "track":
+            return False, True
+
+        (
+            is_premium,
+            premium_conditions,
+            content_owner_id,
+        ) = self._is_content_premium(premium_content_id)
+
+        if not is_premium:
+            # premium_conditions should always be null here as it makes
+            # no sense to have a non-premium track with conditions
+            if premium_conditions:
+                logger.warn(
+                    f"premium_content_access_checker.py | _aggregate_conditions | non-premium content with id {premium_content_id} and type {premium_content_type} has premium conditions."
+                )
+            return False, True
+
+        if not premium_conditions:
+            # is_premium should always be false here as it makes
+            # no sense to have a premium track with no conditions
+            if is_premium:
+                logger.warn(
+                    f"premium_content_access_checker.py | _aggregate_conditions | premium content with id {premium_content_id} and type {premium_content_type} has no premium conditions."
+                )
+            return is_premium, True
+
+        does_user_have_access = self._evaluate_conditions(
+            user_id, cast(int, content_owner_id), premium_conditions
+        )
+        return True, does_user_have_access
+
+    # Returns a tuple of (bool, Dict | None, int | None) -> (track is premium, track premium conditions, track owner id)
+    def _is_content_premium(
```

Comment:
nits: 
- Same comment here re: returning tuples - should opt for an object so caller doesn't have to look up documentation to know what the return type means
- This method doesn't just return whether the content is premium, it returns some other metadata as well so should be named accordingly - `get_premium_content_data` or something?
- This method is tightly coupled to `track`, maybe should have an `PremiumContentType` arg as well?